action.yml

```
@@ -0,0 +1,232 @@
+        if ! [[ "$ENDTOEND_SUITE" =~ ^[a-zA-Z0-9_\-]*$ ]]; then
+          echo "Invalid input for endtoend_suite"
+          exit 1
+        fi
+        if ! [[ "$ENDTOEND_CONFIG" =~ ^[a-zA-Z0-9_\./\-]*$ ]]; then
+          echo "Invalid input for endtoend_config"
+          exit 1
+        fi
+
+    - name: Run PHPUnit
+      # input booleans are converted to strings
+      # https://docs.github.com/en/actions/learn-github-actions/expressions#functions
+      if: ${{ inputs.phpunit == 'true' }}
+      shell: bash
+      env:
+        PHPUNIT_SUITE: ${{ inputs.phpunit_suite }}
+      run: |
+        FAIL_ON_WARNING=
+        if [ "${{ inputs.phpunit_fail_on_warning }}" == "true" ]; then
+          FAIL_ON_WARNING='--fail-on-warning'
+        fi
+        echo "Running PHPUnit with options: $FAIL_ON_WARNING"
+        if [ "$PHPUNIT_SUITE" == "all" ] || [ "$PHPUNIT_SUITE" == "" ]; then
+          vendor/bin/phpunit --verbose --colors=always $FAIL_ON_WARNING
+        else
+          vendor/bin/phpunit --verbose --colors=always --testsuite "$PHPUNIT_SUITE" $FAIL_ON_WARNING
+        fi
+        echo "Passed"
+
+    - name: Setup chrome and chromedriver
+      if: ${{ inputs.endtoend == 'true' }}
+      shell: bash
+      run: |
+        echo "Default versions of google-chrome and chromedriver"
+        GCVR=$(google-chrome --version)
+        CDVR=$(chromedriver --version)
+        echo "$GCVR"
+        echo "$CDVR"
+        # Example version number is 101.0.4951.64
+        [[ "$GCVR" =~ ([0-9]+\.[0-9]+\.[0-9]+)\.[0-9]+ ]]
+        GCV="${BASH_REMATCH[1]}"
+        [[ "$CDVR" =~ ([0-9]+\.[0-9]+\.[0-9]+)\.[0-9]+ ]]
+        CDV="${BASH_REMATCH[1]}"
+        # Reinstall if a.b.c versions do not match, though allow a different .d version
+        if [ "$GCV" != "$CDV" ]; then
+          WGC=$(which google-chrome)
+          echo "google-chrome and chromedriver versions do not match, reinstalling"
+          sudo apt remove -y --purge google-chrome-stable || true
```

What does the `|| true` part do? Is it effectively saying "if there are errors removing chrome, don't fail the job"? If so we should have a comment here to that effect.

---

Don't know, it's copy paste from travis

---

Can you please find out what happens if it's not there so we can document its purpose? It _looks like_ it's intended to hide an error and if we're intentionally hiding errors we should definitely make it clear that's what we're doing.

---

I just removed the `|| true`